### PR TITLE
Create friends page

### LINKED ISSUE #16: Friends Page
As a user
i want to get in touch with my friends
to compare them with me

- [x] Insert header
- [x] Insert ToTop icon
- [x] Insert user item
- [x] Insert navigation
- [x] Create pseudo avatar functionality
- [x] Create add friend modal
- [x] Implement unfollow and follow functionality
- [x] Get the data from the API

### LINKED ISSUE #22: Searchbar Component
As a user
i need a search form
to add my friends to my friend list

- [x] Create the search input
- [x] Make it controlled
- [x] Search in the stats API
- [x] Output user profile component as data
- [x] Debounce fetches

### PR DESCRIPTION
**Pull request by ChristianSeelemann**
Deployment: https://fortknights-page-friend-3kngxm.herokuapp.com/friends

This was a whole chunk of work :)

- Create pseudo avatar functionality
- Create add friend modal
- Search user
- Implement unfollow and follow functionality
- Create API endpoints
- List friends
- Conditional render for 0 friends
- Make to top section scroll dependent
- many little changes

### CODE REVIEW

File: src/app/components/Button/Button.tsx [hunk 1]
```diff
@@ -16,7 +32,21 @@ export default function Button({
 }: ButtonProps): JSX.Element {
   return (
     <button onClick={onClick} className={`${styles.button} ${styles[style]}`}>
-      <span className={styles.icon}>{icon}</span>
+      <span className={styles.icon}>
```
Comment (phlgr) [inline src/app/components/Button/Button.tsx]: The `<span>` should also disappear when no Icon is present

File: src/app/hooks/useFetch.ts [hunk 2]
```diff
@@ -11,7 +11,11 @@ export default function useFetch<T>(url: string): {
   const isLoading = data === null;
 
   function refetch() {
-    fetch(url)
+    fetch(url, {
+      headers: {
+        Authorization: 'e63f4351-b625ddac-254af606-5a2d8ef0',
```
Comment (phlgr) [inline src/app/hooks/useFetch.ts]: This should not be included in the useFetch hook, also this looks like a secret.

File: src/app/pages/Friends/Friends.tsx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+                  onClick={() => {
+                    handleFriendClick(user[0].id);
+                    setModal(true);
+                    setInputValue('');
+                    setUser('');
+                  }}
+                  link="#"
+                  avatar={'../../src/assets/avatars/5.webp'}
+                />
+              )}
```
Comment (phlgr) [inline src/app/pages/Friends/Friends.tsx]: This looks like duplicate code

File: src/server.ts [hunk 4]
```diff
@@ -15,7 +16,7 @@ app.get('/api/lookup/:id', async (request, response) => {
   const { id } = request.params;
   const user = await fetch(`https://fortniteapi.io/v1/lookup?username=${id}`, {
     headers: {
-      Authorization: 'e63f4351-b625ddac-254af606-5a2d8ef0',
```
Comment (phlgr) [inline src/server.ts]: This can still be found in commit history, regenerate this secret!

**Review by phlgr — CHANGES_REQUESTED** [review]

File: src/app/pages/Friends/Friends.tsx (hunk 3 above)
Comment (ChristianSeelemann) [reply src/app/pages/Friends/Friends.tsx]: The buttonStyle and buttonText props are different dependent if the queried user is already a friend or not. But yes. I think its possible to conditional choose these two props instead the complete component.

**Review by phlgr — APPROVED** [review]

**Pull request merged**